### DISCUSSION TITLE
Redirect the client connection from wsgi to asgi

### DISCUSSION THREAD

**Discussion by arun6894** [miguelgrinberg/python-socketio#996] [General]
Currently, I have a system where if the client is connecting from UI the socket server which is running is WSGI.

In python2.7 using **from socketio import SocketIOServer** 

Now I just want to integrate this with python-socketio ASGI. Is there a way we can redirect the connection from a client to the new server of ASGI?

I need the old server and also a new one to compare/test/two switches from one to another if required.

I don't have control of client connections to change their host and port. Want the client connection to trigger the new server with ASGI. Is there a way for it ?

**Comment by miguelgrinberg** [comment]
I don't really understand how I can help with this. Not sure what argument in the client request you want to use to decide which of the servers to connect to, but this is something that you would implement in a proxy server such as nginx, as this package does not have any routing/redirecting capabilities. It's not designed for that.

**Reply by arun6894** [reply]
@miguelgrinberg  I have implemented the socket connection URL through the proxy server pointing to the new server of python-socketio. I could see a 404 status code. It could understand the server but couldn't make connections.


**await self.sio.connect('wss://kd-dsf.system.co.in/test/1/?tab_id=1482&target=table&server_id=ken-server:ff805.system.co.in:43453&session_id=b7a4e9cb-636d-4f36-8c05-b92950f92ef0&uid=35789546-4201-7eee-822b-ec2b94ccb573&time=1660244516609', socketio_path='test/1')**

I have added socketio_path as the polling connection has defaulted to socketi.io
Is it necessary to add /test/1 in the namespace?

**Reply by miguelgrinberg** [reply]
The path and the namespace are completely different things, so no, the namespace should be your chosen namespace, usually `/`. The path is normally `/socket.io`, but you can change it to be whatever you like, as long as you change in the server, the client, and the proxy.

**Reply by arun6894** [reply]
@miguelgrinberg  Yes, in the client I'm requesting with `/game/1`. As per the docs, I could see the below points.

> First, the path is checked against the Socket.IO endpoint.
> Next, the path is checked against the static file configuration, if present.
> If the path did not match the Socket.IO endpoint or any static file, control is passed to the secondary application if configured, else a 404 error is returned.

So we need to configure the static file for the server to understand the new endpoint and only then connection happen without a 404.
??

I have added two lines in server code to match the endpoint from the server. Still, I see 404.
app. router.add_static('/static', 'static')
app. router.add_get('/game/1', index)

**Reply by miguelgrinberg** [reply]
I'm sorry but I don't understand. There are no endpoints for the static file configuration, as it name implies, this is how you serve static files. The documentation shows how to configure serving of static files if that is what you are after.

If you think this package has a bug, please provide a complete application that demonstrates the bug.
